src/modules/roc_pipeline/converter_sink.cpp

```diff
@@ -38,16 +38,18 @@ ConverterSink::ConverterSink(const ConverterConfig& config,
 
         resampler_.reset(audio::ResamplerMap::instance().new_resampler(
                              config.resampler_backend, allocator, config.resampler,
-                             config.input_channels, config.internal_frame_size),
+                             config.internal_frame_length, config.input_sample_rate,
+                             config.input_channels),
                          allocator);
 
         if (!resampler_) {
             return;
         }
 
-        resampler_writer_.reset(new (allocator)
-                                    audio::ResamplerWriter(*awriter, *resampler_, pool,
-                                                           config.internal_frame_size),
+        resampler_writer_.reset(new (allocator) audio::ResamplerWriter(
+                                    *awriter, *resampler_, pool,
+                                    config.internal_frame_length,
+                                    config.output_sample_rate, config.output_channels),
```

Comment:
ResamplerWriter uses frame_length to split its input stream (samples written into it) into frames, so we should use `input_sample_rate` instead of `output_sample_rate` here and in `new_resampler()`.

Reply:
and `output_channels` should be changed to `input_channels` for ResamplerWriter?

Reply:
Yes!

Reply:
done.